src/com/dotmarketing/util/UtilMethods.java

```diff
@@ -1211,21 +1216,42 @@ public static String cleanFileSystemPathURI(String path) {
 
     public static String validateFileName(String fileName) throws IllegalArgumentException{
 
-        if (!isSet(fileName)  ||
-        		fileName.indexOf("..") != -1 || fileName.indexOf("WEB-INF") != -1 || fileName.indexOf("META-INF") != -1 || fileName.indexOf("!") != -1
-                || fileName.indexOf(":") != -1 || fileName.indexOf(";") != -1 || fileName.indexOf(";") != -1 || fileName.indexOf("&") != -1
-                || fileName.indexOf("?") != -1 || fileName.indexOf("$") != -1 || fileName.indexOf("*") != -1 || fileName.indexOf("\"") != -1
-                || fileName.indexOf("/") != -1 || fileName.indexOf("[") != -1 || fileName.indexOf("]") != -1 || fileName.indexOf("=") != -1
-                || fileName.indexOf("|") != -1 || fileName.indexOf(",") != -1 || fileName.indexOf("#") != -1 || fileName.indexOf("{") != -1
-                || fileName.indexOf("}") != -1 || fileName.indexOf("\\") != -1) {
-            throw new IllegalArgumentException("Invalid Filename passed in: " + fileName);
+        if (!isSet(fileName) || REGEX_FILENAME_INVALID_CHARS.matcher(fileName).matches()) {
```

Comment:
I like the way you change it (very smart) however I think this change is a very good candidate to create an unit test to test previous code working against new code.